### PR TITLE
Extracting `test_runner` module

### PR DESCRIPTION
**Pull request by kasperski95**
Conflict-prone PR. Quick review plz.

### CODE REVIEW

**Review by mkaput — COMMENTED** [review]
nitpick: What about `protostar.testing` module? It does not have underscore in names and looks cooler 😅

File: protostar/test_runner/test_environment_exceptions.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+from typing import Any, Dict, List, Optional, Type, TypeVar, Union
 
 from starkware.starknet.business_logic.execution.objects import Event
 from typing_extensions import Literal
 
-from protostar.commands.test.expected_event import ExpectedEvent
 from protostar.starknet.hint_local import HintLocal
 from protostar.utils.log_color_provider import SupportedColorName, log_color_provider
 
+from .expected_event import ExpectedEvent
```
Comment (mkaput) [inline protostar/test_runner/test_environment_exceptions.py:11]: ReportedException, CheatcodeException and some more should go to `protostar.starknet` module. They are also used in the migrator!

File: protostar/test_runner/testing_seed.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 
 def determine_testing_seed(seed: Optional[Seed]) -> Seed:
+
```
Comment (mkaput) [inline protostar/test_runner/testing_seed.py:8]:
```suggestion
```

**Review by mkaput — CHANGES_REQUESTED** [review]

**Review by mkaput — APPROVED** [review]

File: protostar/test_runner/test_environment_exceptions.py (hunk 1 above)
Comment (kasperski95) [reply protostar/test_runner/test_environment_exceptions.py]: Is this resolved?

Comment (kasperski95) [reply protostar/test_runner/test_environment_exceptions.py]: Marking as resolved because PR is approved.

**Pull request merged**